packages/dx-react-grid-bootstrap3/src/plugins/table-view.jsx

```diff
@@ -4,42 +4,53 @@ import { combineTemplates } from '@devexpress/dx-react-core';
 import { TableView as TableViewBase } from '@devexpress/dx-react-grid';
 import { Table } from '../templates/table';
 import { TableCell } from '../templates/table-cell';
-import { TableRow } from '../templates/table-row';
 import { TableStubCell } from '../templates/table-stub-cell';
 import { TableStubHeaderCell } from '../templates/table-stub-header-cell';
 import { TableNoDataCell } from '../templates/table-no-data-cell';
+import { TableRow } from '../templates/table-row';
 
 const tableLayoutTemplate = props => <Table {...props} />;
 const defaultRowTemplate = props => <TableRow {...props} />;
+const defaultNoDataRowTemplate = props => <TableRow {...props} />;
 const defaultCellTemplate = props => <TableCell {...props} />;
 const stubCellTemplate = props => <TableStubCell {...props} />;
 const stubHeaderCellTemplate = props => <TableStubHeaderCell {...props} />;
 const noDataCellTemplate = props => <TableNoDataCell {...props} />;
 
-export const TableView = ({ tableCellTemplate, tableRowTemplate, ...props }) => (
-  <TableViewBase
-    tableLayoutTemplate={tableLayoutTemplate}
-    tableRowTemplate={combineTemplates(
-      tableRowTemplate,
-      defaultRowTemplate,
-    )}
-    tableCellTemplate={combineTemplates(
-      tableCellTemplate,
-      defaultCellTemplate,
-    )}
-    tableStubCellTemplate={stubCellTemplate}
-    tableStubHeaderCellTemplate={stubHeaderCellTemplate}
-    tableNoDataCellTemplate={noDataCellTemplate}
-    {...props}
-  />
+export const TableView = ({
+  tableCellTemplate,
+  tableRowTemplate,
+  tableNoDataRowTemplate,
+  ...props }) => (
+    <TableViewBase
+      tableLayoutTemplate={tableLayoutTemplate}
+      tableRowTemplate={combineTemplates(
+        tableRowTemplate,
+        defaultRowTemplate,
+      )}
+      tableNoDataRowTemplate={combineTemplates(
+        tableNoDataRowTemplate,
+        defaultNoDataRowTemplate,
+      )}
+      tableCellTemplate={combineTemplates(
+        tableCellTemplate,
+        defaultCellTemplate,
+      )}
+      tableStubCellTemplate={stubCellTemplate}
+      tableStubHeaderCellTemplate={stubHeaderCellTemplate}
+      tableNoDataCellTemplate={noDataCellTemplate}
```

Comment:
Why don't we allow to customize three templates above?

Reply:
I'd like to provide this functionality in the separate PR. This PR is about row templates.